topostats/processing.py

```diff
@@ -79,17 +79,21 @@ def run_filters(
             **filter_config,
         )
         filters.filter_image()
-
         # Optionally plot filter stage
         if plotting_config["run"]:
             plotting_config.pop("run")
             LOGGER.info(f"[{filename}] : Plotting Filtering Images")
+            if plotting_config["image_set"] == "all":
+                filter_out_path.mkdir(parents=True, exist_ok=True)
+                LOGGER.debug(f"[{filename}] : Target filter directory created : {filter_out_path}")
             # Generate plots
             for plot_name, array in filters.images.items():
                 if plot_name not in ["scan_raw"]:
                     if plot_name == "extracted_channel":
                         array = np.flipud(array.pixels)
-                    plotting_config["plot_dict"][plot_name]["output_dir"] = filter_out_path
+                    plotting_config["plot_dict"][plot_name]["output_dir"] = (
+                        core_out_path if plotting_config["plot_dict"][plot_name]["core_set"] else filter_out_path
+                    )
                     try:
                         Images(array, **plotting_config["plot_dict"][plot_name]).plot_and_save()
                         Images(array, **plotting_config["plot_dict"][plot_name]).plot_histogram_and_save()
```

Comment:
The only other thing is that you've got an `if image_set=='all'` statement which is then used again within the `plot_and_save()` function, and I'm thinking would it make sense just to use it the once in processing?

We could pop it and re-add it like the `plotting.run` param and keep all the non-core images within this `if` statement, and the core image outside like it is? What do you think?

Reply:
The logic here though is to create the sub-directory `filter` as held in `filter_out_path`. The same is true for `grains`. We then need to set the parameter in the current images `plot_dict` to whatever that is, hence the `if/else`.

When I simplified the logic within `plot_and_save()` for some reason I couldn't understand I got error messages about target directories not being present so I figured I needed to create them.

Reply:
Also having `plot_and_save()` know about `image_set` might be useful should anyone use the class/method interactively e.g. in a Notebook.

Reply:
Ok makes sense, thanks for explaining :)